tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c

```diff
@@ -429,8 +432,9 @@ void test_msgq_full(void)
 	ret = k_sem_init(&end_sema, 0, 1);
 	zassert_equal(ret, 0, NULL);
 
+	ret = k_msgq_put(&msgq1, &data[0], K_NO_WAIT);
+	zassert_equal(ret, 0, NULL);
```

Comment:
Can you please help me understand? This test is put message to a full queue, and should get a -ENOMSG if a timeout is set to K_NO_WAIT (from description). 

We init'd the msgq with size 1, so these lines here are to fill that buffer. Then the next part (k_thread_create) is what executes our test scenario of a thread putting more data on a full queue. Is my understanding correct, or close?

If so, how do we confirm the test scenario was handled properly? Would we not need a `zassert_equal(ret, -ENOMSG, NULL)`?

Reply:
Considering this PR is to fix the unchecked return value, my clarification above should not block this PR.

Reply:
@jenmwms thanks for your review and approval, this is indeed to fix the coverity issue. 

When it first k_msgq_put (see line 435),  if won't get a -ENOMSG because it is still not empty, (msgq->used == 0, msgq->max_msgs==1 ,  as long as  used < max, it will return 0.  see "kernel/msg_q.c" line 126, 148 please.

And your understanding is correct. If we keep calling  k_msgq_put again with K_NO_WAIT, we will get -ENOMSG. like you can see in the created thread (line 231).

Hope this answer you question, thanks!